runtime/gc_glue_java/FlattenedArrayObjectScanner.hpp

```diff
@@ -0,0 +1,258 @@
+	 * increasing significance, and the least significant bit maps to the slot at the returned
+	 * base pointer.
+	 *
+	 * @param[out] scanMap the bit map for the slots contiguous with the returned base pointer
+	 * @param[out] hasNextSlotMap set this to true if this method should be called again, false if this map is known to be last
+	 * @return a pointer to the first slot mapped by the least significant bit of the map, or NULL if no more slots
+	 */
+	virtual fomrobject_t *
+	getNextSlotMap(uintptr_t *slotMap, bool *hasNextSlotMap)
+	{
+		fomrobject_t *result = GC_HeadlessMixedObjectScanner::getNextSlotMap(slotMap, hasNextSlotMap);
+		/* Ignore hasNextSlotMap from HeadLess, we want to always report that there is another element */
+		*hasNextSlotMap = true;
+		if (result == NULL) {
+			/* No more slots in the current element, get the next element of the array */
+			result = _indexableScanner.nextIndexableElement();
+			if (result == NULL) {
+				/* There are no elements in the array */
+				*hasNextSlotMap = false;
+			} else {
+				_mapPtr = result;
+				_endPtr = (fomrobject_t *)((uint8_t*)_mapPtr + _elementSizeWithoutPadding);
+				GC_HeadlessMixedObjectScanner::initialize(_env, _descriptionBasePtr, _leafBasePtr);
+				/* GC_HeadlessMixedObjectScanner::initialize() may setNoMoreSlots(), so set it back to true. 
+				 * We must also return (hasNextSlotMap = true) on top of this
+				 */
+				setMoreSlots();
+			}
+		}
+		return result;
+	}
+
+#if defined(OMR_GC_LEAF_BITS)
+	/**
+	 * Return base pointer and slot bit map for next block of contiguous slots to be scanned. The
+	 * base pointer must be fomrobject_t-aligned. Bits in the bit map are scanned in order of
+	 * increasing significance, and the least significant bit maps to the slot at the returned
+	 * base pointer.
+	 *
+	 * @param[out] scanMap the bit map for the slots contiguous with the returned base pointer
+	 * @param[out] leafMap the leaf bit map for the slots contiguous with the returned base pointer
+	 * @param[out] hasNextSlotMap set this to true if this method should be called again, false if this map is known to be last
+	 * @return a pointer to the first slot mapped by the least significant bit of the map, or NULL if no more slots
+	 */
+	virtual fomrobject_t *
+	getNextSlotMap(uintptr_t *slotMap, uintptr_t *leafMap, bool *hasNextSlotMap)
+	{
+		fomrobject_t *result = GC_HeadlessMixedObjectScanner::getNextSlotMap(slotMap, leafMap, hasNextSlotMap);
+		/* Ignore hasNextSlotMap from HeadLess, we want to always report that there is another element */
+		*hasNextSlotMap = true;
+		if (result == NULL) {
+			/* No more slots in the current element, get the next element of the array */
+			result = _indexableScanner.nextIndexableElement();
+			if (result == NULL) {
+				/* There are no elements in the array */
+				*hasNextSlotMap = false;
+			} else {
+				_mapPtr = result;
+				_endPtr = (fomrobject_t *)((uint8_t*)_mapPtr + _elementSizeWithoutPadding);
```

Comment:
Same as above - change cast to `(uintptr_t)` or add space

Reply:
Fixed